Add variadic Privmsg functions

ripcurld00d opened the pull request:
PrivmsgFmt is the variadic version of the Privmsg function.
It formats the message from a list of arguments before sending it
using the Privmsg function.

For example, the following sends "golang is # 1 IMHO" to #go-nuts:

    c.PrivmsgFmt("#go-nuts", "golang is #", 1, " IMHO")

**Note**: this is a small change, which I find handy.


### EDIT

`Privmsgln` and `Privmsgf` are variadic versions of the `Privmsg` function.
They format a message according to `fmt.Sprintln` and `fmt.Sprintf`
respectively and send it using `Privmsg`.

Note: Although the `Privmsgln` function's name infers that a new-line
character is added at the end of the message, it does not do that.

For example, the following send "golang is # 1 IMHO" to #go-nuts:

    c.Privmsgln("#go-nuts", "golang is #", 1, " IMHO")
    c.Privmsgf("#go-nots", "%s %d %s", "golang is #", 1, "IMHO")

fluffle commented:
Okay, I can see this would be useful.

The thing that bothers me about this is that it uses a different naming convention for functions than fmt itself, where you would have: Privmsg (your current PrivmsgFmt), Privmsgf, and Privmsgln. Given IRCs conventions around line handling Privmsg and Privmsgln would behave identically, so it seems to me that this would better be called Privmsgln. The addition of a Privmsgf might also be useful, if you're feeling so inclined.

ripcurld00d commented:
Good point! Did some changes, let me know what you think!

ripcurld00d commented:
I think we should leave Privmsg as is, because calling `fmt` gives a performance overhead (did anyone say reflection? :wink:) and I don't like it. Nevertheless, you had a good point regarding the name convention. Thus, I have renamed PrivmsgFmt to Privmsgln and added Privmsgf.

What do you say? :smile_cat:

ripcurld00d commented:
I think we should leave Privmsg as is, because calling `fmt` gives a performance overhead (did anyone say reflection? :wink:) and I don't like it. Nevertheless, you had a good point regarding the name convention. Thus, I have renamed PrivmsgFmt to Privmsgln and added Privmsgf.

What do you say? :smile_cat:

fluffle commented:
That's exactly what I was after, thanks a lot :-)
